features/220-cms-import-asset/api.md

```diff
@@ -0,0 +1,183 @@
+# User Requirement
+
+- Import records with skygear asset field
+- Allow cms user to upload file from cms client, when importing records
+- In the csv, the value of asset field would be the file name of uploaded files
+
+# Overview
+
+Summary of design decision here.
+
+### Keep the cms import one single API call
+
+Import feature for each project can be very different if we want to make a
+cutomized cms for the client, due to the different work flow of the client and
+the different target user.
+
+At the current state, CMS provides the import function as a whole feature,
+meaning if the client requires customization beyond what we provided, it is
+likely the developer need to implement one from scratch themselves.
+
+Thus I HIGHLY recommend to keep this feature as simple as possible.
+
+The cms frontend would be responsible for uploading the files and create a
+mapping of file name to asset id for server.
+
+### Accept zip for assets upload
+
+A more natural approach would be to allow cms user to select files to upload.
+However, considering the behaviour of import that if the field value is empty,
+it would replace the original value with null.
+
+Think of this scenario, a cms user import a csv with some files first, then
+the user make some changes to the csv without touching the asset field, if the
+user does not upload the files again, the server would clear the previous value
+of asset field, because the server cannot find the skygear asset.
+
+So to match user expectation, the cms would require users to upload a single csv
+or a zip file that contains all the files.
+
+# UI Update
+
+(The following does not mean the final design)
+
+### Import button clicked
+
+```
+|--------------------------|
+| Import                   |
+|--------------------------|
+| Upload a csv or zip file |
+| here.                    |
+|                          |
+|                 Continue |
+|--------------------------|
+```
+
+- The whole area of `Upload a csv or zip file` is a drag and drop file upload
```

Comment:
I think this issue is about import asset? Or we will have import records and import assets on the same interface? (But I think it would be confusing?)

I guess it is easier for layman users to understand import records and import asset ID as two different things:

1. Import assets: upload a butch of files, we will generate an asset ID and keep it filename somewhere.
2. Import records: upload a CSV or whatever, the asset field should either reference the file by it's asset ID, or filename.

It also imply filename should be unique?